pages/design/test-cases.md

```diff
@@ -0,0 +1,68 @@
+
+- `img` that passes the rule
+- `svg` that passes the rule
+- `img` that fails the rule
+- `svg` that fails the rule
+- some other element that is inapplicable
+
+Providing these test cases ensures that an implementor that correctly tests `img` elements, but passes every `svg` element can be identified as a partial implementation, and one where it fails all `svg` elements is identified as an inconsistent implementation.
+
+## One Thing At A Time
+
+A test case should only test one "condition" at a time. This keeps them focused and easy to reason about. It may be tempting to combine multiple cases in one example, but then if that example does not get the expected outcome, it is difficult to identify the cause of the issue.
+
+When writing test cases, it is important to look at conditional logic. For example if the expectation says an element is applicable if it is "_visible_ and _included in the accessibility tree_", in addition to a passed and failed example, there are three ways for this to not be true, all of which have to be tested:
+
+- inapplicable: visible, but **not** included in the accessibility tree
+- inapplicable: **not** visible, but included in the accessibility tree
+- inapplicable: **not** visible, and **not** included in the accessibility tree
+
+## Consistency With Accessibility Requirements
+
+In order for the mapping of implementations to the rules to work, test cases need to be consistent with the accessibility requirement. This requires that **passed and inapplicable test cases should satisfy the accessibility requirements** of the rule, as well as the accessibility requirement of any composite rule this rule is designed for. Inapplicable conditions that do not satisfy the accessibility requirements should not be included.
+
+Related to this, failed test cases should only fail the accessibility requirements for one reason. A failed test case for 4.1.2 Name, Role, Value should not have issues with both the accessible name, and the role. If the rule is about accessible names, the role and value must be correct.
+
+## Small Test Cases
+
+A general principle in test case creation is to keep test cases small. Avoid any unnecessary content, as it may cause unexpected side effects in the outcome. For example, a rule about buttons does not require that all test cases have an HTML element with a `lang` attribute and a `title` element. Those elements do not affect the accessibility of a button.
+
+An exception to that, is that ACT Rules should strive to show good accessibility practices. A rule about valid `autocomplete` attributes should still provide labels for the form controls with the `autocomplete` attribute. Those labels both provide context, showing the intended purpose of the input, and they show good practice of providing labels for form controls.
+
```

Comment:
Keeping test cases small is a challenging thing to explain. The various "is descriptive" rules do include the HTML element and lang attribute, but none of the other prescribed wrapping elements such as head, title, and body.

To some extent everything specifically included within a test case should be relevant to the rule, and absent wrapping elements can be assumed to follow best practice.

I think it would be good to state that in some way, though I'm not sure that ^^ is the best phrasing.